### PR TITLE
user pdf exports: participant lists, access data

### PR DESCRIPTION
**Pull request by MaximilianKrambach**
User export, mostly reusing the code from 2.3. I have a bit of confusion about the new menu entries in the user list, though (do we need a download dialog here, too?)

### CODE REVIEW

**Comment by emanuelschuetze** [comment]
export should not be a bulk action..

**Comment by emanuelschuetze** [comment]
access data pdf does not work ...

**Comment by emanuelschuetze** [comment]
please rename "Access data (CSV)" to "Export as CSV"

File: client/src/app/core/services/pdf-document.service.ts [hunk 1]
```diff
@@ -446,6 +459,21 @@ export class PdfDocumentService {
...
+            userDataTitle: {
+                fontSize: 26,
+                margin: [0, 0, 0, 0],
+                bold: true
+            },
+            tableHeader: {
+                bold: true,
+                fillColor: 'white'
+            },
+            tableEven: {
```
Comment (tsiegleauq) [inline client/src/app/core/services/pdf-document.service.ts]: We do not need `tableEven` and `tableOdd` anymore. PDFmake can do that now using table style functions.
I comment them bellow.

File: client/src/app/site/users/services/user-pdf.service.ts [hunk 2]
```diff
@@ -0,0 +1,294 @@
...
+     * @param users: The users to include, in order
+     * @returns column definitions with odd and even entries styled differently,
+     * short name, structure level and group name columns
+     */
+    private getListUsers(users: ViewUser[]): object[] {
+        const result = [];
+        let counter = 1;
+        users.forEach(user => {
+            const groupList = user.groups.map(grp => this.translate.instant(grp.name));
+            const tablecolor = counter % 2 === 1 ? 'tableOdd' : 'tableEven';
```
Comment (tsiegleauq) [inline client/src/app/site/users/services/user-pdf.service.ts]: not necessary anymore

File: client/src/app/site/users/services/user-pdf.service.ts [hunk 3]
```diff
@@ -0,0 +1,294 @@
...
+                }
+            ]
+        ];
+        return {
+            table: {
+                widths: ['auto', '*', 'auto', 'auto'],
+                headerRows: 1,
+                body: userTableBody.concat(this.getListUsers(users))
+            },
+            layout: 'headerLineOnly'
```
Comment (tsiegleauq) [inline client/src/app/site/users/services/user-pdf.service.ts]: to get a "zebry style" we might want to provide out own layout function.
That's simpler that the attempt I did choose before with the "tableEven" and "tableOdd" function.

``` ts
layout: {
    fillColor: (rowIndex, node, columnIndex) => {
        return (rowIndex % 2 === 0) ? '#EEEEEE' : null;
    }
}
```

**Comment by MaximilianKrambach** [comment]
done @tsiegleauq

File: client/src/app/site/users/services/user-pdf.service.ts [hunk 4]
```diff
@@ -252,7 +252,11 @@ export class UserPdfService {
                 headerRows: 1,
                 body: userTableBody.concat(this.getListUsers(users))
             },
-            layout: 'headerLineOnly'
+            layout: {
+                fillColor: rowIndex => {
```
Comment (tsiegleauq) [inline client/src/app/site/users/services/user-pdf.service.ts]: better, but has some really big borders.
pleas use that:
```ts
layout: {
    hLineWidth: rowIndex => {
        return rowIndex === 1;
    },
    vLineWidth: () => {
        return 0;
    },
    fillColor: rowIndex => {
        return rowIndex % 2 === 0 ? '#EEE' : null;
    }
}
```

**Comment by tsiegleauq** [comment]
@MaximilianKrambach please resolve the conflicts and rebase into one commit.

**Comment by MaximilianKrambach** [comment]
døne

**Pull request merged**